tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/analytics/AnalyticsServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.dotcms.api.client.analytics;
+
+import com.dotcms.api.AnalyticsAPI;
+import com.dotcms.api.client.model.RestClientFactory;
+import com.dotcms.model.analytics.AnalyticsEvent;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.control.ActivateRequestContext;
+import jakarta.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import org.jboss.logging.Logger;
+
+@ApplicationScoped
+public class AnalyticsServiceImpl implements AnalyticsService {
+
+    static final String COMMAND_EVENT_TYPE = "CMD_EXECUTED";
+
+    private final RestClientFactory clientFactory;
+    private final Logger logger;
+
+    @Inject
+    public AnalyticsServiceImpl(final RestClientFactory clientFactory, final Logger logger) {
+        this.clientFactory = clientFactory;
+        this.logger = logger;
+    }
+
+    @Override
+    @ActivateRequestContext
+    public void recordCommand(final String command, final List<String> arguments)
```

Comment:
Shouldn't this be asynchronous?  Perhaps we should wrap this within a Future Call so it doesn't impact execution

Reply:
We call the `AnalyticsServiceImpl` using futures from the `DotExecutionStrategy`.